openhands/sdk/conversation/state.py

```diff
@@ -48,3 +81,94 @@ def __exit__(self, exc_type, exc, tb) -> None:
     def assert_locked(self) -> None:
         if self._owner_tid != get_ident():
             raise RuntimeError("State not held by current thread")
+
+    # ===== Internal FS helpers (intentionally simple) =====
+
+    @staticmethod
+    def _scan_events(fs: FileStore) -> list[EventFile]:
+        """
+        Single directory listing for events. Returns (index, path) sorted.
+        """
+        try:
+            paths = fs.list(EVENTS_DIR)
+        except Exception:
+            return []
+        out: list[EventFile] = []
+        for p in paths:
+            name = p.rsplit("/", 1)[-1]
+            m = _EVENT_NAME_RE.match(name)
+            if m:
+                out.append(EventFile(idx=int(m.group("idx")), path=p))
+            else:
+                logger.warning(f"Skipping unrecognized event file: {p}")
+        out.sort(key=lambda t: t.idx)
+        return out
+
+    @staticmethod
+    def _replay_from_files(
```

Comment:
```suggestion
    def _restore_from_files(
```
Nit: we are using the same term in `v0` too, but maybe we can use `restore` here? Because `replay` functionality exists, and it means something else, the replaying of events in the runtime, e.g. re-executing things in a new container. On the other hand, `restore` is used with the meaning of reloading, or restoring session, which is similar with what happens here.